### PR TITLE
[Feat.] Autoscaling: Add Autoscaling Lifecycle hooks

### PR DESCRIPTION
**Pull request by muneeb-jan**
## Summary of the Pull Request
Adding the feature [autoscaling life hooks](https://docs.otc.t-systems.com/auto-scaling/umn/as_management/scaling_action/managing_lifecycle_hooks.html) to the Terraform provider. 

## PR Checklist

* [x] Refers to: #2656 
* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.
* [x] Release notes added.

## Acceptance Steps Performed

```
=== RUN   TestAccASV1LifecycleHook_basic
=== PAUSE TestAccASV1LifecycleHook_basic
=== CONT  TestAccASV1LifecycleHook_basic
--- PASS: TestAccASV1LifecycleHook_basic (84.63s)
PASS
Process 421623 has exited with status 0
```

### CODE REVIEW

File: opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+		UpdateContext: resourceASLifecycleHookUpdate,
+		DeleteContext: resourceASLifecycleHookDelete,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ForceNew: true,
+			},
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go]: please put region to computed, not needed for create request

File: opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+				ValidateFunc: resourceASLifeCycleHookValidateName,
+			},
+			"scaling_lifecycle_hook_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: false,
+				ValidateFunc: validation.StringInSlice([]string{
+					"INSTANCE_TERMINATING", "INSTANCE_LAUNCHING",
+				}, true),
+				Description: "Specifies the lifecycle hook type. Options: INSTANCE_TERMINATING, INSTANCE_LAUNCHING",
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go]: remove description, we store parameter description only in doc

File: opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+			},
+			"default_result": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: false,
+				ValidateFunc: validation.StringInSlice([]string{
+					"ABANDON", "CONTINUE",
+				}, true),
+				Description: "Specifies the default lifecycle hook callback operation. Options: CONTINUE, ABANDON(Default)",
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go]: and here

File: opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+				Default:     "ABANDON",
+			},
+			"default_timeout": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      3600,
+				ValidateFunc: validation.IntBetween(60, 86400),
+				Description:  "Timeout duration, in seconds.",
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go]: here too

File: opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go [hunk 5]
```diff
@@ -0,0 +1,239 @@
...
+				ForceNew:     false,
+			},
+			"notification_topic_urn": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    false,
+				Description: "URN of the SMN Topic",
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go]: here

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+)
+
+func ResourceASLifecycleHook() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceASLifecycleHookCreate,
+		ReadContext:   resourceASLifecycleHookRead,
+		UpdateContext: resourceASLifecycleHookUpdate,
+		DeleteContext: resourceASLifecycleHookDelete,
+
+		Schema: map[string]*schema.Schema{
```
Comment (anton-sidelnikov) [inline opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go:33]: also, please add importer and check import in test

**Review by anton-sidelnikov — CHANGES_REQUESTED** [review]

File: opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go (hunk 1 above)
Comment (muneeb-jan) [reply opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go]: It is already set to computed.

Comment (anton-sidelnikov) [reply opentelekomcloud/services/as/resource_opentelekomcloud_as_lifecycle_hook_v1.go]: I mean remove optional and force new

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**